src/mds/FSMap.h

```diff
@@ -259,17 +259,21 @@ class FSMap {
       return;
     }
 
-    for (auto &f : filesystems) {
-      std::string_view fs_name = f.second->mds_map.get_fs_name();
+    for(auto it = filesystems.begin(); it != filesystems.end();) {
```

Comment:
nit, might want to define a helper like
```c++
#if __cplusplus <= 201703L
template<class Key, class T, class Compare, class Alloc, class Pred>
typename std::map<Key, T, Compare, Alloc>:size_type
erase_if(std::map<Key, T, Compare, Alloc>& c, Pred pred) {
  auto old_size = c.size();
  for (auto i = c.begin(), last = c.end(); i != last; ) {
    if (pred(*i)) {
      i = c.erase(i);
    } else {
      ++i;
    }
  }
  return old_size - c.size();
}
#endif
```

so we can use it before we have the luxury of using C++20.

Reply:
please have a look at a fix: https://github.com/ibm-s390-cloud/ceph/commit/3ae4069a9f7586db50dc659e98420af36015113d

Reply:
You would want refresh this PR with ibm-s390-cloud@3ae4069 ?